### PR TITLE
Support to load query packs as scheduled queries

### PR DESCRIPTION
**Pull request by javuto**
This code allows loading query packs from the configuration, as the element "packs" and pointing to a different file per pack.

### CODE REVIEW

**Comment by javuto** [comment]
ok to test

**Comment by theopolis** [comment]
The build linking errors are because I installed a static libcryptsetup on those boxes and you now need to include gcrypt and devmapper as dynamic links in the CMakeLists.txt for tables. We can get around that by (1) reverting all machines back to 0 and letting them rebuild from scratch or, (2) landing https://github.com/facebook/osquery/pull/1141 then `git pull --rebase origin master` on this and all other PRs.

File: osquery/config/parsers/query_packs.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ public:
+  /// Request "packs" top level key.
+  std::vector<std::string> keys() { return {"packs"}; }
+
+ private:
+  /// Store the signatures and file_paths and compile the rules.
+  Status update(const std::map<std::string, ConfigTree>& config);
+};
+
+pt::ptree QueryPackSingleEntry(const pt::ptree pack_data) {
```
Comment (theopolis) [inline osquery/config/parsers/query_packs.cpp]:
```cpp
pt::ptree QueryPackSingleEntry(const pt::ptree& pack_data) {
```

File: osquery/config/parsers/query_packs.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    // Iterate through all the packs to get the configuration
+    auto pack_name = std::string(pack_element.first.data());
+    auto pack_path = std::string(pack_element.second.data());
+
+    // Read each pack configuration in JSON
+    pt::ptree pack_tree;
+    status = osquery::parseJSON(pack_path, pack_tree);
+
+    if (!status.ok()) {
+      LOG(ERROR) << "Error parsing Pack JSON : " << status.getCode() << " - " << status.getMessage();
```
Comment (theopolis) [inline osquery/config/parsers/query_packs.cpp]: `ERROR` or `WARNING` ?

File: osquery/config/parsers/query_packs.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      return status;
+    }
+
+    // Get all the parsed elements from the pack JSON file
+    pt::ptree pack_file_element = pack_tree.get_child(pack_name);
```
Comment (theopolis) [inline osquery/config/parsers/query_packs.cpp]: This might not work (may throw an exception), you might have to `pack_tree.count(pack_name)`

**Comment by theopolis** [comment]
Some tests and you're gtg

File: osquery/config/parsers/query_packs.cpp (hunk 2 above)
Comment (javuto) [inline osquery/config/parsers/query_packs.cpp]: ```WARNING``` should be fine

**Comment by theopolis** [comment]
Build should now succeed, ok to test

**Pull request merged**